src/queueFsm.js

```diff
@@ -74,13 +74,17 @@ var Factory = function( options, connection, topology, serializers, queueFn ) {
       }.bind( this );
 
       var subscriber = function( exclusive ) {
-        queue.subscribe( !!exclusive )
-          .then(
-            onSubscribe,
-            function( err ) {
-              emit( "subscribeFailed", err )
-            }
-          );
+		return when.promise( function( resolve, reject ) {
+			queue.subscribe( !!exclusive )
+			  .then(
+				onSubscribe,
+				function( err ) {
+				  emit( "subscribeFailed", err )
+				  reject("subscribeFailed " + err);
+				}
+			  )
+			  .then(resolve);
+		});
```

Comment:
Thanks for your insight, what was i thinking, updated the commit to reflect the changes